alma/controllers/hook/FrontHeaderHookController.php

```diff
@@ -64,74 +162,138 @@ private function handleOrderPage($params)
+    }
+
+    /**
+     * Manage assets for Prestashop Before 1.7.
+     *
+     * @return string
+     */
+    protected function manageAssetVersionForPrestashopBefore17()
+    {
+        $content = '';
+
+        $this->controller->addJS(ConstantsHelper::WIDGETS_JS_URL);
+        $this->controller->addJS($this->module->_path . ConstantsHelper::PRODUCT_SCRIPT_PATH);
+
+        if ($this->displayWidgetOnCartPage()) {
+            $this->controller->addJS($this->module->_path . ConstantsHelper::CART_SCRIPT_PATH);
+        }
+
+        if ($this->displayWidgetOnProductPage()) {
+            $this->controller->addCSS($this->module->_path . ConstantsHelper::PRODUCT_CSS_PATH);
+        }
+
+        if (version_compare(_PS_VERSION_, '1.5.6.2', '<')) {
+            $content = '<link rel="stylesheet" href="' . ConstantsHelper::WIDGETS_CSS_URL . '">';
+        }
+
+        if (version_compare(_PS_VERSION_, '1.5.6.2', '>=')) {
+            $this->controller->addCSS(ConstantsHelper::WIDGETS_CSS_URL);
+        }
+
+        return $content;
     }
 
-    private function injectAlmaAssets($params)
+    /**
+     * Manage assets for Prestashop after 1.7
+     *
+     * @return string
+     */
+    protected function manageAssetVersionForPrestashopAfter17()
     {
-        $widgetsCssUrl = 'https://cdn.jsdelivr.net/npm/@alma/widgets@3.x.x/dist/widgets.min.css';
-        $widgetsJsUrl = 'https://cdn.jsdelivr.net/npm/@alma/widgets@3.x.x/dist/widgets.umd.js';
-        $productScriptPath = 'views/js/alma-product.js';
-        $productCssPath = 'views/css/alma-product.css';
-        $cartScriptPath = 'views/js/alma-cart.js';
+        $content = '';
+
+        $scriptPath = "modules/$this->moduleName/" . ConstantsHelper::PRODUCT_SCRIPT_PATH;
+        $cssPath = "modules/$this->moduleName/" . ConstantsHelper::PRODUCT_CSS_PATH;
+        $cartScriptPath = "modules/$this->moduleName/" . ConstantsHelper::CART_SCRIPT_PATH;
+
+        $this->controller->registerStylesheet(ConstantsHelper::PRODUCT_CSS_ID, $cssPath);
+
+        if ($this->displayWidgetOnCartPage()) {
+            $this->controller->registerJavascript(ConstantsHelper::CART_SCRIPT_ID, $cartScriptPath, ['priority' => 1000]);
+        }
+
+        if ($this->displayWidgetOnProductPage()) {
+            $this->controller->registerJavascript(ConstantsHelper::PRODUCT_SCRIPT_ID, $scriptPath, ['priority' => 1000]);
+        }
+
+        if (version_compare(_PS_VERSION_, ConstantsHelper::PRESTASHOP_VERSION_1_7_0_2, '<')) {
+            // For versions 1.7.0.0 and 1.7.0.1, it was impossible to register a remote script via FrontController
+            // with the new registerJavascript method, and the deprecated addJS method had been changed to be just a
+            // proxy to registerJavascript...
+            $content = <<<TAG
+                    <link rel="stylesheet" href="{${ConstantsHelper::WIDGETS_CSS_URL}}">
+                    <script src="{${ConstantsHelper::WIDGETS_JS_URL}}"></script>
+TAG;
```

Comment:
For my information, this indentention is ok?

Reply:
Yes, the TAG add space in the code between the balise 😅